plugin-server/src/worker/ingestion/person-state.ts

```diff
@@ -543,55 +550,83 @@ export class PersonState {
         // This is low-probability so likely won't occur on second retry of this block.
         // In the rare case of the person changing VERY often however, it may happen even a few times,
         // in which case we'll bail and rethrow the error.
-        await this.db.postgresTransaction('mergePeople', async (client) => {
-            try {
-                const [person, updatePersonMessages] = await this.db.updatePersonDeprecated(
-                    mergeInto,
-                    {
-                        created_at: firstSeen,
-                        properties: excludeProperties
-                            ? mergeInto.properties
-                            : this.updatedPersonProperties(mergeInto.properties),
-                        is_identified: true,
-                    },
-                    client
-                )
+        try {
+            let mergedPerson
+                // Keep the oldest created_at (i.e. the first time we've seen this person)
+            ;[kafkaMessages, mergedPerson] = await this.handleMergeTransaction(
+                mergeInto,
+                otherPerson,
+                olderCreatedAt,
+                properties
+            )
+            await this.db.kafkaProducer.queueMessages(kafkaMessages)
 
-                // :KLUDGE: Avoid unneeded fetches in updateProperties()
-                this.personContainer = this.personContainer.with(person)
+            // :KLUDGE: Avoid unneeded fetches in updateProperties()
+            this.personContainer = this.personContainer.with(mergedPerson)
```

Comment:
I recall now why it was outside of the catch block...
Note: if either of these failed we'll also retry the merge, which could be fine 🤷‍♀️

Reply:
So what will happen if postgres transaction succeeded but we failed sending kafka messages is that we're losing out on those kafka messages as rerunning aliasDeprecated will see that the persons are already the same and do nothing. I.e. CH <-> Postgres out of sync problem this PR isn't making it worse in any way as that happened the same before.